### DISCUSSION TITLE
Is it possible to allow the last task in a DAG to fail, but for the DAG Run to still be successful?

### DISCUSSION THREAD

**Discussion by pedro-cf** [apache/airflow#43443] [Q&A]
Is it possible to allow the last task in a DAG to fail, but for the DAG Run to still be successful?

Without adding an Empty Operator at the end with trigger rule "all_done"?
Or without adding a try/except in the task code?

Some kind of soft_fail but for the standard @task ?

**Comment by potiuk** [comment]
You should look at setup/teardown functionality (added in 2.8 or 2.9) https://airflow.apache.org/docs/apache-airflow/stable/howto/setup-and-teardown.html - the teardown task is what you are searching for.

**Reply by pedro-cf** [reply]
I'll explain my scenario.

I have a specific notification task that runs at the end of the DAG.

This notification task must only run if the other tasks succeed  (with teardown it would always run?).

On the odd chance this notification task fails I don't want the DAG Run to fail.

**Reply by potiuk** [reply]
You can always extend the notifcation to always succeed, regardless if it fails.

**Reply by pedro-cf** [reply]
> You can always extend the notifcation to always succeed, regardless if it fails.

I'm sorry but I don't know what you mean by this?

**Reply by potiuk** [reply]
```
try:
  your code for notification
except Exception:
  pass
```

You can write your own operator, (see custom operators in the docs) or write task flow tasks that will NEVER fail, regardless if the code to execute it failed or not. That's easy. you want the task to succeed, when your notification failed. So make it succeed.
